src/service.ts

```
@@ -47,6 +67,22 @@ export class SolanaService extends Service {
+    this.jupiterService = this.runtime.getService(serviceType) as any;
+    new Promise(async resolve => {
+      while (!this.jupiterService) {
+        console.log(asking, 'waiting for', serviceType, 'service...');
+        this.jupiterService = this.runtime.getService(serviceType) as any;
+        if (!this.jupiterService) {
+          await new Promise((waitResolve) => setTimeout(waitResolve, 1000));
+        } else {
+          console.log(asking, 'Acquired', serviceType, 'service...');
+        }
+      }
+    })
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)